python/cucumber_expressions/expression.py

```diff
@@ -87,20 +86,34 @@ def rewrite_alternation(self, node: Node):
     def rewrite_alternative(self, node: Node):
         return "".join([self.rewrite_to_regex(_node) for _node in node.nodes])
 
-    def rewrite_parameter(self, node: Node):
+    def rewrite_parameter(self, node: Node) -> str:
         name = node.text
-        parameter_type = self.parameter_type_registry.lookup_by_type_name(name)
+        group_name, parameter_type = self.parse_parameter_name(name)
 
         if not parameter_type:
             raise UndefinedParameterTypeError(node, self.expression, name)
 
-        self.parameter_types.append(parameter_type)
+        self.parameter_types_and_names.append((parameter_type, group_name))
 
         regexps = parameter_type.regexps
         if len(regexps) == 1:
             return rf"({regexps[0]})"
         return rf"((?:{')|(?:'.join(regexps)}))"
 
+    def parse_parameter_name(
+        self, name: str
+    ) -> Tuple[Optional[str], Optional[ParameterType]]:
+        """Helper function to parse the parameter name and return group_name and parameter_type."""
+        if ":" in name:
+            group_name, parameter_type_name = [part.strip() for part in name.split(":")]
```

Comment:
This allows for the empty group name, which is distinct from the `None` group name. Probably now what we want.

Reply:
It might also be worth while to push this into the parser.